flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/ExecNodeContext.java

```diff
@@ -104,9 +104,16 @@ private ExecNodeContext(@Nullable Integer id, String name, Integer version) {
     @JsonCreator
     public ExecNodeContext(String value) {
         this.id = null;
-        String[] split = value.split("_");
-        this.name = split[0];
-        this.version = Integer.valueOf(split[1]);
+        try {
+            String[] split = value.split("_");
+            this.name = split[0];
+            this.version = Integer.valueOf(split[1]);
+        } catch (Exception e) {
+            throw new TableException(
```

Comment:
Not sure if this generic catch all is helpful. Esp instructing users to file a ticket although we already know that we serialized an invalid node. Should we explicitly check for `name == "null"` and tell users that the node is unsupported?

Reply:
Initially I also thought of just null checking. I went for `try catch` instead of a null check, because theoretically there might be a few reasons for an exception:
1. Error while splitting.. we don't have two parts
2. Error while parsing the Integer, the second part is not a number
3. Error while parsing the Integer, the part is a null

I understand 1 & 2 are unlikely atm, but I guess we thought the same about the `null_null` situation.

Reply:
The null_null situation was a bug in the serialization method. During JSON deserialization so many things can go wrong if users modify the JSON plan themselves. I vote for only implement what we need for this particular use case. Also adding a comment there, why we need to to support plans older than 1.18.1.